core/controllers/get_filters.py

```diff
@@ -66,19 +66,27 @@ def get_outcome_categories():
     return outcome_categories, 200
 
 
-def get_regions():
-    """Returns all unique itl region codes associated with projects.
+def get_geospatial_regions():
+    """Returns all unique ITL1 region codes associated with projects.
 
-    :return: a list of itl region codes
+    :return: a list of ITL1 region codes and region names in dictionaries
     """
-    projects = Project.query.all()
-
-    itl_regions = set(region for project in projects for region in project.itl_regions)
+    geospatial_itl1_regions = (
+        GeospatialDim.query.order_by(GeospatialDim.data_blob["itl1_region_name"])
+        .distinct(GeospatialDim.itl1_region_code, GeospatialDim.data_blob["itl1_region_name"])
+        .with_entities(GeospatialDim.data_blob, GeospatialDim.itl1_region_code)
+        .where(GeospatialDim.projects != None)  # noqa
+        .all()
+    )
 
-    if not itl_regions:
+    if not geospatial_itl1_regions:
         return abort(404, "No regions found.")
 
-    return list(itl_regions), 200
+    itl_regions = [
```

Comment:
is this supposed to return a list of simple one-line dicts? Like a JSON? Why not just a single dict containing all the lines? with key = name and val = code? or is this what the frontend code needs? If keeping the former, maybe update the docstring description to reflect this?

Reply:
Yeah it returns a list of dictionaries with the ITL1 region code as id and region name as name, eg. `[{"id": "TLI", "name": "London"}, {"id": "TLN", "name": "Northern Ireland"}]` which is what the frontend needs to iterate through to create a checkbox for each element in the list and grab the ID when the user selects a checkbox.

Will update the docstring as per your other comment to make reflect this (I think this old get filter was never used and as it was implemented it would just return a list of the unique ITL1 region codes in use, but the front-end still used its own hardcoded list and this endpoint was never called).